src/Mod/TechDraw/Gui/QGIView.cpp

```diff
@@ -244,30 +233,39 @@ void QGIView::mousePressEvent(QGraphicsSceneMouseEvent * event)
 {
 //    Base::Console().Message("QGIV::mousePressEvent() - %s\n",getViewName());
     signalSelectPoint(this, event->pos());
+    if (m_dragState == NODRAG) {
+        m_dragState = DRAGSTARTED;
+    }
 
     QGraphicsItem::mousePressEvent(event);
 }
 
-//void QGIView::mouseMoveEvent(QGraphicsSceneMouseEvent * event)
-//{
-//    QGraphicsItem::mouseMoveEvent(event);
-//}
+void QGIView::mouseMoveEvent(QGraphicsSceneMouseEvent * event)
+{
+    if (m_dragState == DRAGSTARTED) {
+        m_dragState = DRAGGING;
+    }
+    QGraphicsItem::mouseMoveEvent(event);
+}
 
 void QGIView::mouseReleaseEvent(QGraphicsSceneMouseEvent * event)
 {
     //TODO: this should be done in itemChange - item position has changed
-    //TODO: and should check for dragging
 //    Base::Console().Message("QGIV::mouseReleaseEvent() - %s\n",getViewName());
 //    if(scene() && this == scene()->mouseGrabberItem()) {
-    if(!m_locked) {
-        if (!isInnerView()) {
-            double tempX = x(),
-                   tempY = getY();
-            getViewObject()->setPosition(Rez::appX(tempX),Rez::appX(tempY));
-        } else {
-            getViewObject()->setPosition(Rez::appX(x()),Rez::appX(getYInClip(y())));
+    if (m_dragState == DRAGGING) {
+        if(!m_locked) {
+            if (!isInnerView()) {
+                double tempX = x(),
+                       tempY = getY();
+                getViewObject()->setPosition(Rez::appX(tempX),Rez::appX(tempY));
+            } else {
+                getViewObject()->setPosition(Rez::appX(x()),Rez::appX(getYInClip(y())));
+            }
         }
     }
```

Comment:
```suggestion
    if (m_dragState == DRAGGING && !m_locked) {
        if (!isInnerView()) {
            double tempX = x(),
                    tempY = getY();
            getViewObject()->setPosition(Rez::appX(tempX),Rez::appX(tempY));
        } else {
            getViewObject()->setPosition(Rez::appX(x()),Rez::appX(getYInClip(y())));
        }
    }
```

Reply:
But why?  I find the original much easier to understand.  Optimizing compilers are our friends.